src/daemon/Display.cpp

```diff
@@ -466,10 +467,6 @@ namespace SDDM {
         // greeter
         if (status != Auth::HELPER_AUTH_ERROR)
             stop();
-
-        // Start the greeter again as soon as the user session is closed
```

Comment:
so what happens now when a user does log out in a root-full X session?

Reply:
Display::stop gets called (right before the removal). This informs Seat that we've stooped and we go to Seat::displayStopped.

This removes the display and if it's display-less (which is the case as we just removed the display) it creates a new one which will start a new greeting process.